### ISSUE TITLE
Job summary on PR

### ISSUE THREAD

**Issue by samueljsmalmeida** [cypress-io/github-action#1058]
Could have an option to publish the job summary to the PR please.

**Comment by MikeMcC399** [comment]
@samueljsmalmeida 

> Could have an option to publish the job summary to the PR please.

I understand that you are asking for a feature enhancement to the action. Could you explain a little more what you would like and how it would work?

Please also describe why this is needed and what benefit it would give.

**Comment by samueljsmalmeida** [comment]
Many thanks for the very quick response.

The general idea is to have an option to publish the test summary to the respective PR as a comment.

If the parameter `publish-summary-to-pr` is set to true, and if the `github.event_name == 'pull_request'`, then add the test summary MD as a comment to the respective PR.

The benefit of said feature is to deliver the option to centralize relevant info on the PR and avoid having to go to the Actions tab.

Many thanks

**Comment by MikeMcC399** [comment]
@samueljsmalmeida 

It's an interesting idea. I guess we have to wait to see what the Cypress team thinks and if anybody is prepared to submit a PR to implement it.

The best alternative at the moment is to view the results in the PR, for example https://github.com/cypress-io/github-action/pull/1059 which I submitted today shows:

![image](https://github.com/cypress-io/github-action/assets/66998419/82f3a35d-8aaf-4ac3-8167-da9ae02d86d7)

however clicking on any of the relevant Details link like https://github.com/cypress-io/github-action/actions/runs/6559572703/job/17815434926?pr=1059 does not show the Job Summary

![image](https://github.com/cypress-io/github-action/assets/66998419/2624c488-70b4-4b50-88ff-6490b828c8d3)

Only if you click on the Summary over on the left do you then get to the desired overview page

![image](https://github.com/cypress-io/github-action/assets/66998419/7fb75909-1c0f-4fb4-92bf-a3fa5afc61ce)

**Comment by samueljsmalmeida** [comment]
Exactly, my idea is to avoid having to navigate to the Action summary.
Thanks

**Comment by albertodeago** [comment]
Would love to see this too

**Comment by ghiscoding** [comment]
I also wish that this action could add the summary into the PR, that is exactly what Codecov is doing and it is super convenient to get all the info directly in the PR (see below). 

Another approach could be to use this other GitHub Action [EnricoMi/publish-unit-test-result-action](https://github.com/EnricoMi/publish-unit-test-result-action) that collects any test reports and displays the summary into multiple places (as a PR comment, as a Job Summary, ...) but that requires us to run Cypress with reporters as JUnit or JSON which is then passed to that `publish-unit-test-result-action` action. I tried to give it a go but at the end it requires so much setup and external dependencies that I gave up (it shows below, as the last summary, but with 0 tests), however I do have it working with Playwright in another project.

At the end of the day, Cypress already have all the info displayed in the Job Summary, it's just a matter of sending it to the PR as well. Note that it might need extra permissions to write to the PR but that's just a setup that user could accept to opt-in or not (e.g. `publish-unit-test-result-action` requires these permissions  `checks: write` and `pull-requests: write`). If Codecov can do it (which is always does by default, no config required), I'm sure Cypress can do it too

![image](https://github.com/cypress-io/github-action/assets/643976/0f64e06a-fa02-4543-98fc-43f8e53bfc58)

**Comment by krainboltgreene** [comment]
Honestly just adding the grid to the output would be a huge boon, then people could do the commenting bit themselves.

**Comment by lumattr** [comment]
If you could have the job summaries of all the jobs displayed at the bottom of the PR, that could negate the need for having to comment at all. My issue at the moment is that we have seperate workflows that are trying to publish information to the PR but we end up with comment spam because, whilst they each use a comment maintain action. Theres still 5 comments for each PR. Job summaries would be good, but with the details link going to the job page rather than summary, i have to guide the devs to the output data.

**Comment by mroy-seedbox** [comment]
Got here from Google search because we are also looking for the exact same feature (which in the end sounds like a GitHub feature request instead):

> If you could have the job summaries of all the jobs displayed at the bottom of the PR, that could negate the need for having to comment at all. [...] Job summaries would be good, but with the details link going to the job page rather than summary, i have to guide the devs to the output data.

I'm also not sure why the link sends to the logs/details instead of the job summary... The summary would be a much more helpful starting point (and from there we can dig into the details _if necessary_).

**Comment by dwjohnston** [comment]
I think this is the relevant issue for what I'm currently facing. 

I'm using Cypress Cloud and what I want is that for each test run, a link to the Cypress Cloud run should be included _somewhere_ in the PR. 

We could do this by setting up a Github Action that comments the link in the PR - but how to get the link? 

We can grep it from std out - the bit that says: `  Recorded Run: https://cloud.cypress.io/projects/xyz/runs/16`, but how to get access to this? 

If we do something like


```diff
      - name: Execute Cypress
        uses: cypress-io/github-action@v6
        with:
          command: npx cypress run | tee cypress-output.log
          wait-on: http://localhost:8080
          record: true
```

The problem with this is that it that the additional commands (--record) will be tacked on to `tee` not `npx`.

**Comment by samueljsmalmeida** [comment]
@dwjohnston , I would probably start by looking at this Action https://github.com/marketplace/actions/comment-pull-request
